python/tests/unit/arcticdb/version_store/test_lazy_dataframe.py

```diff
@@ -0,0 +1,456 @@
+    lazy_df = lazy_df[lazy_df["col1"] <= 7]
+    received = lazy_df.collect().data
+    expected = df.iloc[6:8]
+
+    assert_frame_equal(expected, received)
+
+
+def test_lazy_apply(lmdb_library):
+    lib = lmdb_library
+    sym = "test_lazy_apply"
+    df = pd.DataFrame(
+        {"col1": np.arange(10, dtype=np.int64), "col2": np.arange(100, 110, dtype=np.int64)}, index=pd.date_range("2000-01-01", periods=10)
+    )
+    lib.write(sym, df)
+
+    lazy_df = lib.read(sym, lazy=True)
+    lazy_df = lazy_df.apply("new_col", lazy_df["col1"] + lazy_df["col2"])
+    received = lazy_df.collect().data
+    expected = df
+    expected["new_col"] = expected["col1"] + expected["col2"]
+
+    assert_frame_equal(expected, received)
+
+
+def test_lazy_apply_inline_col(lmdb_library):
+    lib = lmdb_library
+    sym = "test_lazy_apply_inline_col"
+    df = pd.DataFrame(
+        {"col1": np.arange(10, dtype=np.int64), "col2": np.arange(100, 110, dtype=np.int64)}, index=pd.date_range("2000-01-01", periods=10)
+    )
+    lib.write(sym, df)
+
+    lazy_df = lib.read(sym, lazy=True).apply("new_col", col("col1") + col("col2"))
+    received = lazy_df.collect().data
+    expected = df
+    expected["new_col"] = expected["col1"] + expected["col2"]
+
+    assert_frame_equal(expected, received)
+
+
+def test_lazy_project(lmdb_library):
+    lib = lmdb_library
+    sym = "test_lazy_project"
+    df = pd.DataFrame(
+        {"col1": np.arange(10, dtype=np.int64), "col2": np.arange(100, 110, dtype=np.int64)}, index=pd.date_range("2000-01-01", periods=10)
+    )
+    lib.write(sym, df)
+
+    lazy_df = lib.read(sym, lazy=True)
+    lazy_df["new_col"] = lazy_df["col1"] + lazy_df["col2"]
+    received = lazy_df.collect().data
+    expected = df
+    expected["new_col"] = expected["col1"] + expected["col2"]
+
+    assert_frame_equal(expected, received)
+
+
+def test_lazy_groupby(lmdb_library):
+    lib = lmdb_library
+    sym = "test_lazy_groupby"
+    df = pd.DataFrame({"col1": [0, 1, 0, 1, 2, 2], "col2": np.arange(6, dtype=np.int64)})
+    lib.write(sym, df)
+
+    lazy_df = lib.read(sym, lazy=True)
+    lazy_df = lazy_df.groupby("col1").agg({"col2": "sum"})
+    received = lazy_df.collect().data
+    received.sort_index(inplace=True)
+    expected = df.groupby("col1").agg({"col2": "sum"})
+
+    assert_frame_equal(expected, received)
+
+
+def test_lazy_resample(lmdb_library):
+    lib = lmdb_library
+    sym = "test_lazy_resample"
+    df = pd.DataFrame(
+        {"col1": np.arange(10, dtype=np.int64), "col2": np.arange(100, 110, dtype=np.int64)}, index=pd.date_range("2000-01-01", periods=10)
+    )
+    lib.write(sym, df)
+
+    lazy_df = lib.read(sym, lazy=True)
+    lazy_df = lazy_df.resample("D").agg({"col1": "sum", "col2": "first"})
+    received = lazy_df.collect().data
+    expected = df.resample("D").agg({"col1": "sum", "col2": "first"})
+
+    assert_frame_equal(expected, received)
+
+
+def test_lazy_with_initial_query_builder(lmdb_library):
```

Comment:
What's the motivation for this usage? Should we just forbid it and simplify things a bit?

Reply:
I guess it doesn't really add anything with `read`. With `read_batch` I can see it being used when you have one query you want to apply to all symbols before splitting to apply separate queries to different symbols, and I want to keep `read` and `read_batch` consistent with each other.